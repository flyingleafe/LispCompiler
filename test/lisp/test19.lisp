(define factorial (lambda (n)
                    (let ((fac (lambda (acc n)
                                 (if (= n 0)
                                     acc
                                   (recur (* acc n) (- n 1))))))
                      (fac 1 n))))

(printString "Enter the value:")
(printInt (factorial (readInt)))
