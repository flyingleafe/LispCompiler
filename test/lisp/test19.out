Enter the value:
720
